core/src/main/scala/org/apache/spark/rdd/RDDBarrier.scala

```diff
@@ -54,5 +54,27 @@ class RDDBarrier[T: ClassTag] private[spark] (rdd: RDD[T]) {
     )
   }
 
+  /**
+   * :: Experimental ::
+   * Returns a new RDD by applying a function to each partition of the wrapped RDD,
```

Comment:
adding "while tracking the index of the original partition" to make it the same as the doc in ```RDD.mapPartitionsWithIndex```?